tests/tracer/test_writer.py

```diff
@@ -497,6 +498,8 @@ def test_flush_connection_reset(endpoint_test_reset_server):
         writer._send_payload("foobar", 12)
 
 
+# This test is flaky on CircleCI and raises a BrokenPipeError.
+@flaky.flaky(max_runs=5)
```

Comment:
do you have any thoughts on how we can make this not flaky? rather than automatically rerunning it?

Reply:
Actually, I was not able to reproduce the issue locally so I assume it is transient and/or related to the CI reliability. Flagging the test as flaky allows to ignore transient errors but still fail the test if the issue persists.

Reply:
I am ok with this mark as flaky/rerun process, as long as we have a process in place to review the known flaky tests to try and resolve.

Reply:
On master we now have the retry logic to deal with such cases. Perhaps we could call `flush_queue` in the test instead of `_send_payload` directly?

Reply:
Indeed there is a retry logic in the writer, I will try to use it instead.